contracts/strategies/ltip-simple/LTIPSimpleStrategy.sol

```diff
@@ -0,0 +1,666 @@
+        }
+
+        // Check if the metadata is required and if it is, check if it is valid, otherwise revert
+        if (metadataRequired && (bytes(metadata.pointer).length == 0 || metadata.protocol == 0)) {
+            revert INVALID_METADATA();
+        }
+
+        // If the recipient address is the zero address this will revert
+        if (recipientAddress == address(0)) revert RECIPIENT_ERROR(recipientId);
+
+        // Get the recipient
+        Recipient storage recipient = _recipients[recipientId];
+
+        // update the recipients data
+        recipient.recipientAddress = recipientAddress;
+        recipient.useRegistryAnchor = registryGating ? true : isUsingRegistryAnchor;
+        recipient.allocationAmount = allocationAmount;
+        recipient.metadata = metadata;
+        recipient.recipientStatus = Status.Pending;
+
+        if (recipient.recipientStatus == Status.None) {
+            // If the recipient status is 'None' add the recipient to the '_recipientIds' array
+            _recipientIds.push(recipientId);
+            emit Registered(recipientId, _data, _sender);
+        } else {
+            emit UpdatedRegistration(recipientId, _data, _sender);
+        }
+    }
+
+    /// @notice Allocate (delegated) voting power to a recipient. In the simple strategy, every authorized voter has 1 vote.
+    /// @dev '_sender' must be a pool manager to allocate.
+    /// @param _data The data to be decoded
+    /// @param _sender The sender of the allocation
+    function _allocate(bytes memory _data, address _sender)
+        internal
+        virtual
+        override
+        nonReentrant
+        onlyActiveAllocation
+    {
+        // Decode the '_data'
+        address recipientId = abi.decode(_data, (address));
+        Recipient storage recipient = _recipients[recipientId];
+
+        if (recipient.recipientStatus != Status.Accepted) revert RECIPIENT_NOT_ACCEPTED();
+
+        // Check if the allocator has already casted the vote
+        address voteCastedTo = votedFor[_sender];
+        if (voteCastedTo != address(0)) {
+            // remove the old vote to allow recasting of vote
+            votes[voteCastedTo] -= 1;
+        }
+
+        // Increment the votes for the recipient
+        votes[recipientId] += 1;
+        // Update the votedFor mapping
+        votedFor[_sender] = recipientId;
+
+        // Emit the event
+        emit Voted(recipientId, _sender);
+
+        if (votes[recipientId] >= votingThreshold) {
+            emit Allocated(recipientId, recipient.allocationAmount, allo.getPool(poolId).token, address(0));
+        }
+    }
+
+    function _vestAmount(address recipientId, address recipientAddress, address _token, uint256 _amount)
+        internal
+        virtual
+    {
+        TokenTimelock vestingContract =
+            new TokenTimelock(IERC20(_token), recipientAddress, block.timestamp + vestingPeriod);
+
+        IAllo.Pool memory pool = allo.getPool(poolId);
```

Comment:
This is already available on the BaseStrategy.sol so you can avoid the call

Reply:
Which one, to pool.token?? I did notice that the call for `_token` is redundant because it's in the params, but could't find it in the BaseStrategy?

Updating to simplify it to:

```
    function _vestAmount(address recipientId, address recipientAddress, address _token, uint256 _amount)
        internal
        virtual
    {
        TokenTimelock vestingContract =
            new TokenTimelock(IERC20(_token), recipientAddress, block.timestamp + vestingPeriod);

        _transferAmount(_token, address(vestingContract), _amount);

        _vestingPlans[recipientId] = VestingPlan(address(vestingContract), 0);

        emit VestingPlanCreated(recipientId, address(vestingContract), 0);
    }
    ```

Reply:
As we got a thumbsup to submit for review, resolving this convo